src/test/java/com/smartbear/readyapi/client/DataGenDataSourceTest.java

```diff
@@ -0,0 +1,79 @@
+package com.smartbear.readyapi.client;
+
+import com.smartbear.readyapi.client.model.BooleanDataGenerator;
+import com.smartbear.readyapi.client.model.DataGenDataSource;
+import com.smartbear.readyapi.client.model.DataSourceTestStep;
+import com.smartbear.readyapi.client.model.TestStep;
+import com.smartbear.readyapi.client.teststeps.TestStepTypes;
+import org.junit.Test;
+
+import static com.smartbear.readyapi.client.TestRecipeBuilder.newTestRecipe;
+import static com.smartbear.readyapi.client.teststeps.TestSteps.dataGenDataSource;
+import static com.smartbear.readyapi.client.teststeps.datasource.datagen.DataGenerators.booleanTypeProperty;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+public class DataGenDataSourceTest {
+
+    @Test
+    public void buildsRecipeWithDataSourceTestStepWithBooleanDataGenDataSource() throws Exception {
+        TestRecipe recipe = newTestRecipe()
+                .addStep(dataGenDataSource()
+                        .withNumberOfRows(34)
+                        .addDataGenerator(
```

Comment:
That will be difficult since the return type will be different (builder classes are different for dataGenDataSource and generators), so the context will not be same.